### ISSUE TITLE
msodbcsql17 for Debian version 11

### ISSUE THREAD

**Issue by JiriPrajz** [MicrosoftDocs/sql-docs#7255]
Please add information for install msodbcsql17 under Debian 11. 

Thanks Jiri Prajz

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 0fc3da20-7ab2-d20f-4d29-4664d3183b57
* Version Independent ID: 986d25d9-5d2e-5154-39ea-3b683faf2264
* Content: [Install the Microsoft ODBC driver for SQL Server (Linux) - ODBC Driver for SQL Server](https://docs.microsoft.com/en-us/sql/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server?view=sql-server-ver15)
* Content Source: [docs/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server.md](https://github.com/MicrosoftDocs/sql-docs/blob/live/docs/connect/odbc/linux-mac/installing-the-microsoft-odbc-driver-for-sql-server.md)
* Product: **sql**
* Technology: **connectivity**
* GitHub Login: @David-Engel
* Microsoft Alias: **v-davidengel**

**Comment by WilliamAntonRohm** [comment]
@JiriPrajz -- Jiri, thank you for your feedback.  Please consider these resources:
- [SQL Server on Microsoft Q&A](https://docs.microsoft.com/answers/products/sql-server)
- [DBA Stack Exchange](https://dba.stackexchange.com/questions/tagged/sql-server)
- [Stack Overflow](https://stackoverflow.com/questions/tagged/sql-server)

@David-Engel -- David, please look into this issue.

**Comment by David-Engel** [comment]
Debian 11 installation instructions will be added when msodbcsql17 supports. Debian 11.

Regards,
David

#please-close

**Comment by potiuk** [comment]
Hello @David-Engel @WilliamAntonRohm.

I am one of the maintainers of Apache Airflow: https://github.com/apache/airflow - one of the most popular data workflow orchestrator out there and I would love to get more information about when the Debian 11 will come for msodbcsql17.
Is there a timeline available for that ? How can I subscribe (is there an issue I can watch?)  to information on when Debian 11 will be supported? 

A bit of context:

Lack of `msodbcsql17` is the last thing that is missing for Airflow to move to bullseye as base Docker image to support (note that official Python images have been already for quite some time based on Bullseye, we kept buster support for quite long but we are about to switch). And mssqlodbc17 is actually the last dependency that lags behind and prevents us from migrating to Bullseye. We have > 500 dependencies in total so this is not easy for us to migrate.

It is a bit of Irony for us to be blocked on this move by MSSQL  because we have been planning ([this is experimental for now](https://github.com/apache/airflow#requirements)) to release official MsSQL support for Apache Airflow in the upcoming 2.3.0 release.  That would be first official release of Apache Airflow that would support MsSQL. That was an effort of many people and about 7-8 months of testing the stability and making sure that all our DB use is MsSQL-compliant and now it's blocking us from supporting the most common and popular Debian stable release. It seems that our decision to include it caused this "lagging behind" as mssqlodbc17 driver is the only reliable way we can connect to MSSQL from SQLAlchemy. And we cannot test in CI the Bullseye which we have planned for some time (and this is the default platfform our users are targetting too mostly).

Just to give a bit of context on Bullseye vs. Buster - it's been 6 months after Bullseye has been released and about 6 monthis till the end-of-life of Buster (~2022-08) - https://wiki.debian.org/DebianReleases so - for us at least - it's highest time to move to Bullseye.

If the release is in a week or so, we might still be able to do it in 2.3.0 (which is coming in ~ 1/2 months) but - if this will take longer -  we might consider delaying the "official" support (and possibly disable MSSQL tests and postpone the MSSQL support to much later - we can only add support for new MINOR release which might happen 5-6 months from now. 

I'd love to hear what your plans are in this matter, so that we can make the right decisions for our project.

We (the maintainers of Airflow) are looking forward to hearing from you.

**Comment by David-Engel** [comment]
@potiuk Our regular Jan 31 release was delayed until Feb 15. Watch for the release announcement here: https://techcommunity.microsoft.com/t5/sql-server-blog/bg-p/SQLServer/label-name/SQLServerDrivers

It includes Debian 11 support.

Regards,
David

**Comment by potiuk** [comment]
Fantastic news ! thanks!
